pkg/workceptor/remote_work.go

```diff
@@ -68,17 +75,32 @@ func (rw *remoteUnit) connectToRemote(ctx context.Context) (net.Conn, *bufio.Rea
 }
 
 // getConnection retries connectToRemote until connected or the context expires
-func (rw *remoteUnit) getConnection(ctx context.Context) (net.Conn, *bufio.Reader) {
+func (rw *remoteUnit) getConnection(mw *utils.JobContext) (net.Conn, *bufio.Reader) {
```

Comment:
I don't see where we're making use of any JobContext functionality, so I think this parameter should still be a `context.Context`

Reply:
we call `mw.Cancel()` in this method, line 98